### PR TITLE
`Lectures`: Allow students to view learning goals and prerequisites

### PR DESCRIPTION
**Pull request by ge65cer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Students were only able to see the learning goals on the Statistics page. They had no way of checking the prerequisites for a course.

### Description

With this PR (follow-up for #4950) students can now view the prerequisites for a course when registering for it.
We also introduce a new tab to the course overview page for learning goals, were both the learning goals and prerequisites are visible to students.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with Learning Goals and Prerequisites
- 1 Course with Prerequisites and enabled registration (requires that your user is part of the course's organization)

1. Log in to Artemis as Student
2. Navigate to Course Overview
3. Checkout the added Learning Goals page
4. Go to Course Registration
5. Click on the Prerequisites button to open the modal

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Functions | Line |
|------------|-------:|-----:|
| course-prerequisites-modal.component.ts | 87.5% | 94.4% |
| course-learning-goals.component.ts | 87.5% | 96.1% |

### Screenshots

Screenshot 2022-05-17 at 22 52 21

Screenshot 2022-05-17 at 23 16 00

### CODE REVIEW

**Review by nadinesamyy — APPROVED** [review]
tested on ts1, everything works okay

**Review by TheTrueVester — DISMISSED** [review]
Tested on ts1. Everything worked as expected.

**Review by Dominik-Weinzierl — CHANGES_REQUESTED** [review]
Could you please add some comments in the client code :D Thank you!

**Review by nairaarnez — DISMISSED** [review]
Tested in ts1, it is loking good!

**Review by nairaarnez — APPROVED** [review]

File: src/main/webapp/i18n/de/global.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
                 "modelingExercise": "Modeling Exercise",
                 "quizExercise": "Quiz Exercise",
                 "lecture": "Vorlesungen",
+                "learningGoal": "Lernziele",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/global.json:29]: Even though the JSON ist only a goal (no plural), the translation is plural

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Only found one nitpick about the translation keys.
Besides that, everything is working on TS1

File: src/test/javascript/spec/component/learning-goals/course-learning-goals.component.spec.ts [hunk 2]
```diff
@@ -125,7 +130,8 @@ describe('CourseLearningGoals', () => {
         courseLearningGoalsComponentFixture.detectChanges();
 
         const learningGoalCards = courseLearningGoalsComponentFixture.debugElement.queryAll(By.directive(LearningGoalCardStubComponent));
-        expect(learningGoalCards).toHaveLength(2);
+        expect(learningGoalCards).toHaveLength(3); // 1 prerequisite and 2 learning goals
+        expect(getAllPrerequisitesForCourseSpy).toHaveBeenCalledTimes(1);
```
Comment (Dominik-Weinzierl) [inline src/test/javascript/spec/component/learning-goals/course-learning-goals.component.spec.ts:134]: same as above

File: src/test/javascript/spec/component/course-registration/course-registration.component.spec.ts [hunk 3]
```diff
@@ -85,4 +95,14 @@ describe('CourseRegistrationComponent', () => {
...
         expect(component.coursesToSelect).toEqual([course2]);
     });
+
+    it('should open modal with prerequisites for course', () => {
+        component.coursesToSelect = [course1];
+        const openModalSpy = jest.spyOn(modalService, 'open');
+
+        component.showPrerequisites(course1.id!);
+
+        expect(openModalSpy).toHaveBeenCalledTimes(1);
```
Comment (Dominik-Weinzierl) [inline src/test/javascript/spec/component/course-registration/course-registration.component.spec.ts:105]: You could also use `toHaveBeenCalledOnce()`

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good to me and thank you for adding the comments :D - just two small suggestions that you can decide whether you want to change them or not.

File: src/main/webapp/i18n/de/global.json (hunk 1 above)
Comment (ge65cer) [reply src/main/webapp/i18n/de/global.json:29]: It's the same for lectures and exercises, that's why I kept it consistent.

**Review by TheTrueVester — APPROVED** [review]
Retested on ts1. Everything still works as expected.

**Review by nairaarnez — APPROVED** [review]
Tested in Ts2, it works as described!

**Review by jschmautzer — APPROVED** [review]
tested on ts1, followed the testing steps, worked as expected :)

**Review by babyygemperor — APPROVED** [review]
Tested on ts1, works as described

**Pull request merged**